loopchain/blockchain/blockchain.py

```diff
@@ -1176,19 +1169,19 @@ def _process_next_prep_legacy(self, _block: Block, block_builder: BlockBuilder,
             if change_reason == NextRepsChangeReason.TermEnd:
                 next_leader = ExternalAddress.empty()
 
-            next_preps_hash = Hash32.fromhex(next_prep["rootHash"], ignore_prefix=True)
+            next_preps = [ExternalAddress.fromhex(prep["id"]) for prep in next_prep["preps"]]
+            next_preps_hash = None  # to rebuild next_reps_hash
         else:
             # P-Rep list has no changes
-            next_leader = _block.header.next_leader
+            next_preps = reps
```

Comment:
Does it fit the v0.3 block specification? see. https://icon-project.atlassian.net/wiki/spaces/LO/pages/92045385/Block+0.3